src/Moryx.Communication.Mqtt.Endpoints/ResourceRpcEndpoint.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2025, Phoenix Contact GmbH & Co. KG
+// Licensed under the Apache License, Version 2.0
+
+using Microsoft.Extensions.Logging;
+using Moryx.AbstractionLayer.Identity;
+using Moryx.AbstractionLayer.Resources;
+using Moryx.Communication.Mqtt.Builders;
+using Moryx.Communication.Mqtt.Components;
+using Moryx.Communication.Mqtt.Converters;
+using Moryx.Serialization;
+using MQTTnet;
+
+namespace Moryx.Communication.Mqtt.Endpoints;
+
+/// <summary>
+/// Endpoint to remote call a procedure/method on a resource
+/// </summary>
+public class ResourceRpcEndpoint(
+    ILogger<ResourceRpcEndpoint> logger,
+    IResourceManagement resourceManagement,
+    MqttClientUserOptions options)
+    : IMqttEndpoint
+{
+    public void Map(IMqttRouteBuilder routeBuilder)
+    {
+        routeBuilder.MapGet("resources/{identifier}/invoke/{methodName}", context =>
+        {
+            var messageBuilder = new MqttApplicationMessageBuilder();
+            messageBuilder.WithTopic(context.RequestMessage.ResponseTopic);
+
+            var identifier = context.FromParameterValues<string>("identifier");
+            var methodName = context.FromParameterValues<string>("methodName") ?? string.Empty;
+            var resource = resourceManagement.GetResource<IResource>(x =>
+                x is IIdentifiableObject obj && obj.Identity.Identifier == identifier
+                || x.Name == identifier);
+            return FindAndInvokeMethod(context, messageBuilder, identifier, methodName, resource);
+        });
+    }
+
+    private MqttApplicationMessage? FindAndInvokeMethod(
+        MqttEndpointContext context,
+        MqttApplicationMessageBuilder responseBuilder,
+        string? identifier,
+        string methodName,
+        IResource? resource)
+    {
+        var methodToInvoke = resource
+                        ?.GetType()
+                        .GetInterfaces()
+                        .Select(i => i.GetMethod(methodName))
+                        .FirstOrDefault(x => x is not null);
+
+        if (resource is null || methodToInvoke is null)
+        {
+            logger.Log(LogLevel.Error, "No resource or method found with the given Identifier:`{id}`, method:`{method}`.", identifier, methodName);
+            return null;
+        }
+
+        var methodToInvokeParameters = methodToInvoke.GetParameters();
```

Comment:
You could split the parameter parsing into a separate private method to improve readability and maybe return early if a parameter could not be properly parsed 🤔